mnefun/_mnefun.py

```diff
@@ -3036,3 +3036,87 @@ def annotate_head_pos(raw, head_pos, rotation_limit=45, translation_limit=0.1,
+        Use mininum norm, dSPM (default) or sLORETA.
+    lambda2 : float
+        The regularization parameter.
+    pick_ori : None | “normal” | “vector”
+        If “normal”, rather than pooling the orientations by taking the norm,
+        only the radial component is kept. This is only implemented when
+        working with loose orientations. If “vector”, no pooling of the
+        orientations is done and the vector result will be returned in the
+        form of a mne.VectorSourceEstimate object. This is only implemented
+        when working with loose orientations.
+    n_smooth : int or None
+        Number of iterations for the smoothing of the surface data. If None,
+        smooth is automatically defined to fill the surface with non-zero
+        values.
+    vertices_to : list of arrays of int
+        Vertices for each hemisphere (LH, RH) for subject_to.
+
+    """
+    if subjects_dir is None:
+        subjects_dir = mne.utils.get_subjects_dir(p.subjects_dir,
+                                                  raise_error=True)
+    vertices_to = [np.arange(vertices_to), np.arange(vertices_to)]
+    fname_data = op.join(p.work_dir, analysis + 'fs_data.npz')
+    naves = np.zeros(len(subjects), int)
+    for si, (subj, struc) in enumerate(zip(subjects, structurals)):
+        print('Loading data for subject %s...' % subj)
+        inv_dir = op.join(p.work_dir, subj, p.inverse_dir)
+        s_name = safe_inserter(p.inv_names[0], subj)
+        temp_name = s_name + ('-%d' % p.lp_cut) + p.inv_tag
+        inv_name = op.join(inv_dir, temp_name + '-%s' % inverse + '-inv.fif')
+        # load the inverse
+        inv = read_inverse_operator(inv_name)
+        evoked_fnames = get_epochs_evokeds_fnames(p, subj,
+                                                  analyses)[1]
+        evoked_fname = op.join(inv_dir, '%s_55-sss_eq_%s-ave.fif'
+                        % (analysis, subj))
+        aves = [mne.Evoked(fname, cond, proj=True,
+                           kind='average') for cond in conds]
+        nave = np.unique([a.nave for a in aves])
+        assert len(nave) == 1
+        naves[si] = nave[0]
+
+        # apply inverse, bin, morph
+        stcs = [apply_inverse(ave, inv, lambda2, 'dSPM') for ave in aves]
+        # stcs = [stc.bin(0.02) for stc in stcs]
+        m = mne.compute_morph_matrix(struc, 'fsaverage', stcs[0].vertices,
+                                     fs_verts, n_smooth)
+        stcs = [stc.morph_precomputed('fsaverage', fs_verts, m)
+                for stc in stcs]
+
+        # put in big matrix
+        if subj == subjects[0]:
+            data = np.empty((len(stcs), len(subjects), stcs[0].shape[0],
+                             stcs[0].shape[1]))
+        for di, stc in enumerate(stcs):
+            data[di, si, :, :] = stc.data
+            times = stc.times
+    print('Writing data...')
+    np.savez_compressed(fname_data, data=data, times=times, naves=naves)
+    h5f = h5py.File(fname_data, 'w')
```

Comment:
I wouldn't write it this way, but rather using `stc.save(...)` as it's more standard

Reply:
We could save group averaged data as an `stc` but this writes out individual source data as an ndarray that can later be used for t-testing.

Reply:
I'm saying that individual files should be written for each subject. When t-testing is done you can read data looping over subjects to construct this `ndarray`. It's less convenient but sticks with standard MNE formats. You can always add something like this outside of the `mnefun` scripts. That's how I usually do stats.

Reply:
Oh... that's how I do it already too. Lets just close.